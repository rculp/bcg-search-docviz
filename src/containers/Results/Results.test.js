import { shallow } from 'enzyme';
import uuid from 'uuid/v1'; jest.mock('uuid/v1', () => (jest.fn(() => 1)));

import React from 'react';
import configureStore from 'redux-mock-store';
import { Message } from 'semantic-ui-react';

import ConnectedResultsContainer, { ResultsContainer } from './Results';

describe('Results', () => {
  const mockStore = configureStore();
  const initialState_full = {
    search: {
      results: {
        Result: {
          Boxes: [
            { type: 'TagCloud' },
            { type: 'Tree' },
            { type: 'List' },
            { type: 'Unrecognized Type' }
          ],
          Docs: [
            { filename: 'filename1', largesummaryhtml: '<b>test</b>' }
          ]
        }
      }
    }
  };
  const initialState_noResult = {
    search: {
      results: {
        Result: null
      }
    }
  };
  const initialState_noDocsReturned = {
    search: {
      results: {
        Result: {
          Docs: []
        }
      }
    }
  };
  const mockHistory = {push: jest.fn()};
  let store;

  beforeEach(() => {
    mockHistory.push.mockReset();
    store = mockStore(initialState_full);
  });

  it('connects to redux store', () => {
    const component = shallow(
      <ConnectedResultsContainer store={store} />
    );
    expect(component.props().results).toEqual(initialState_full.search.results);
  });

  it('renders and matches our snapshot', () => {
    const component = shallow(
      <ResultsContainer results={initialState_full.search.results} history={mockHistory}/>
    );
    expect(component).toMatchSnapshot();
  });

  it('shows message if no docs are found', () => {
    const component = shallow(
      <ResultsContainer results={initialState_noDocsReturned.search.results} history={mockHistory}/>
    );
    expect(component.find(Message).props().header).toEqual('No Results Found');
  });

  it('redirects to / if no results are found', () => {
    const component = shallow(
      <ResultsContainer results={initialState_noResult.search.results} history={mockHistory}/>
    );
    expect(mockHistory.push).toHaveBeenCalledWith('/');
  });
});
