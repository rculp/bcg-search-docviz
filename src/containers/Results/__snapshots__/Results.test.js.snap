// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Results renders and matches our snapshot 1`] = `
<Page
  id="results"
>
  <Header />
  <main>
    <Grid>
      <Row>
        <Col
          xs={4}
        >
          <Header
            as="h1"
          >
            Refiners
          </Header>
          <RefinerTagCloud
            box={
              Object {
                "type": "TagCloud",
              }
            }
            key="1"
          />
          <RefinerTree
            box={
              Object {
                "type": "Tree",
              }
            }
            key="1"
          />
          <RefinerList
            box={
              Object {
                "type": "List",
              }
            }
            key="1"
          />
          <div
            key="1"
          >
            Unrecognized refiner type
          </div>
        </Col>
        <Col
          xs={8}
        >
          <Header
            as="h1"
          >
            Results
          </Header>
          <Grid
            key="1"
          >
            <Row>
              <Col
                xs={6}
              >
                filename1
              </Col>
              <Col
                xs={6}
              >
                <b
                  key="0"
                >
                  test
                </b>
              </Col>
            </Row>
          </Grid>
        </Col>
      </Row>
    </Grid>
  </main>
  <Footer />
</Page>
`;
